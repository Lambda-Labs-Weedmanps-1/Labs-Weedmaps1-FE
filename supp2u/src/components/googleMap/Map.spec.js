import React from 'react';
import { mount, configure, shallow, enzyme } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Adapter from 'enzyme-adapter-react-16';

import Map1 from './Map'

configure({ adapter: new Adapter() });

describe('Map Display', () => {
    // If Map component render is ever non functional, this should fail
    it('should render correctly when component called', () => {
      const mappy = shallow(<Map1 />);
    
      expect(mappy).toMatchSnapshot();
    });
});
 
// ERROR: this one causes warning about ACT issues and memory leaks, can't pin down yet
// FIXED [PARTIAL]: cannot used a component called MAP in the tests, it thinks 
// it's calling a function.
test('valid path should yield 1 component, no sub comps for map', async () => {
    // making sure this only has 1 component, not 0 on render of path
    const wrapper = mount(
    <MemoryRouter initialEntries={[ '/home' ]}>
      <Map1 />
    </MemoryRouter>
    );
    await wrapper;
    expect(wrapper.find(Map1)).toHaveLength(1);
  
    wrapper.unmount();
});
