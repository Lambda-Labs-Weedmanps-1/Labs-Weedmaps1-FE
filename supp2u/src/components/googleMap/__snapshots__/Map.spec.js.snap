// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Display should render correctly when component called 1`] = `
<div
  className="landing-container"
>
  <div
    className="landing-blurb"
  >
    <p>
      Connect with your favorite eateries and discover the best culinary experiences in town.
    </p>
    <img
      alt="food image"
      height="50%"
      src="suppburger.jpg"
      width="90%"
    />
  </div>
  <GoogleMaps
    positions={
      Array [
        Object {
          "lat": 0,
          "lng": 0,
        },
      ]
    }
  />
</div>
`;
